subsys/net/lib/http/http_client.c

```diff
@@ -38,6 +38,18 @@ static int sendall(int sock, const void *buf, size_t len)
 			return -errno;
 		}
 
+		if (out_len == 0 && len != 0) {
+			struct zsock_pollfd pfd;
+			int pollres;
+
+			pfd.fd = sock;
+			pfd.events = ZSOCK_POLLOUT;
+			pollres = zsock_poll(&pfd, 1, CONFIG_HTTP_CLIENT_SOCKET_POLL_TIMEOUT_MS);
+			if (pollres > 0) {
```

Comment:
@rlubos What would you suggest on 0 (timeout) and -1 (errors)?
I'd say in case of timeout we proceed (just fall through) and in case of -1 return `errno`?

Reply:
Sounds fine.